pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -384,7 +384,7 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 	if !controllerutils.HasFinalizer(cd, hivev1.FinalizerDeprovision) {
 		cdLog.Debugf("adding clusterdeployment finalizer")
 		if err := r.addClusterDeploymentFinalizer(cd); err != nil {
-			cdLog.WithError(err).Error("error adding finalizer")
+			cdLog.WithError(err).Log(controllerutils.LogLevel(err), "error adding finalizer")
```

Comment:
@staebler can you put the example of log before and after this change. I am not sure what is the effect of `cdLog.WithError` in the logs.

Reply:
If the error is not a conflict error, then the log entry emitted is exactly the same.
If the error is a conflict error, then the only difference in the log entry omitted is that the level is info instead of error.

`WithError(err)` is identical to `WithField("error", err)`